src/components/Main/Main.js

```diff
@@ -11,20 +11,32 @@ export default function Main() {
       uniqueGenreArray.push(movie.genre.split("|")[0]);
     }
   });
+  // {Genre1: [{movie1}, {movie2},
+  // Genre2: [{movie1}{movie2}{movie3}]
+  // ]}
   let sortedGenreMovies = {};
   uniqueGenreArray.forEach((uniqueGenre) => {
     let alikeGenreMovies = [];
     movieData.forEach((movie) => {
       if (movie.genre === uniqueGenre) {
-        alikeGenreMovies.push(movie.title);
+        alikeGenreMovies.push(movie);
         sortedGenreMovies[uniqueGenre] = alikeGenreMovies;
       }
     });
   });
```

Comment:
Making some good progress here. Let's see if we can improve the performance. Right now, its looping through each genre, and within that loop, its looping through all the movies. This means we're going to be looping through all the movies several times. If there's 7 generes, and 100 movies, we will loop 7 * 100 (700 times). This is known as `O(n*m)` https://www.freecodecamp.org/news/big-o-notation-why-it-matters-and-why-it-doesnt-1674cfa8a23c/

Before these loops, we're creating the uniqueGenreArray by looping over the movies. This is creating that array of unique genre's but we're actually already pretty close to creating `sortedGenreMovies` object that we want, we just need some tweaks.

So starting from what you have:
```js
  let uniqueGenreArray = [];
  movieData.forEach((movie) => {
    if (
      !uniqueGenreArray.some((genre) => genre === movie.genre.split("|")[0])
    ) {
      uniqueGenreArray.push(movie.genre.split("|")[0]);
    }
  });
  ```
  
We can jump right to creating the object here,  instead of creating `uniqueGenreArray`.

We can do this by looping over the movies like you are already doing. But, instead of trying to push it into an array, we will use the genre as a key and the value will be the array of movies:

```js
  let sortedGenreMovies = {};
  movieData.forEach((movie) => {
      // First we get the movie genre
      const movieGenre = movie.genre.split("|")[0];
      // Now lets get the current value for this genre.
      // It might be undefined if this is a new genre, or it might be an array
      // Since we need an array, we can use || to assign an empty array if its undefined
      let currentMoviesForGenre = sortedGenreMovies[key] || [];
      // Now we can push our movie into the array
      currentMoviesForGenre.push(movie);
      // And finally set this array as the value for the genre
      sortedGenreMovies[key] = currentMoviesForGenre;
    }
  });
```
This is O(n). Meaning if there are 7 genres, and 100 movies, we only loop 100 times